dart/math/MathTypes.h

```diff
@@ -83,8 +85,28 @@ typedef Matrix<double, 6, 6> Matrix6d;
 typedef std::vector<Eigen::Vector3d> EIGEN_V_VEC3D;
 typedef std::vector<std::vector<Eigen::Vector3d > > EIGEN_VV_VEC3D;
 
+template <typename _Tp>
+using aligned_vector = std::vector<_Tp, Eigen::aligned_allocator<_Tp>>;
+
+template <typename _Key, typename _Tp, typename _Compare = std::less<_Key>>
+using aligned_map = std::map<_Key, _Tp, _Compare,
+    Eigen::aligned_allocator<std::pair<const _Key, _Tp>>>;
+
+template<typename _Tp, typename... _Args>
+inline std::shared_ptr<_Tp> make_aligned_shared()
```

Comment:
> What happens when we make the no-argument version of Eigen::make_aligned_shared() use std::allocate_shared<_Tp>( Eigen::aligned_allocator<_Tp>() )? Does the same error happen?

Yes. It seems the variadic template function generates exactly same function with no-argument version of `Eigen::make_aligned_shared()` when there is no argument, which should do.

I tracked down the history of Eigen's `aligned_allocator` since the error is caused in it. I found that the `Eigen::aligned_allocator` is not compatible with C++11 yet. Here are the related issues: [[1]](http://eigen.tuxfamily.org/bz/show_bug.cgi?id=455), [[2]](http://eigen.tuxfamily.org/bz/show_bug.cgi?id=503).

There as an intermediate patch for this in Eigen 3.1.0 but removed in Eigen 3.2.1, and the new patch is applied to the default branch but not to any released version yet. Ubuntu 14.04 is fine since it has Eigen 3.1.0. But Ubuntu 14.10 and 15.04 causes compilation error since they have Eigen 3.2.1 and Eigen 3.2.2, respectively.

I copied new aligned_allocator from Eigen's default branch and made DART to use it if the installed Eigen is greater than 3.2.1. This should be removed once the new `aligned_allocator` is released.